src/transaction/hooks/useSendWalletTransactions.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import type { UseSendWalletTransactionsParams } from '../types';
+import { sendBatchedTransactions } from '../utils/sendBatchedTransactions';
+import { sendSingleTransactions } from '../utils/sendSingleTransactions';
+
+// Sends transactions to the wallet using the appropriate hook based on Transaction props and wallet capabilities
+export const useSendWalletTransactions = ({
+  capabilities,
+  sendCallAsync,
+  sendCallsAsync,
+  transactions,
+  transactionType,
+  walletCapabilities,
+  writeContractAsync,
+  writeContractsAsync,
+}: UseSendWalletTransactionsParams) => {
+  return useCallback(async () => {
+    if (!transactions) {
+      return;
+    }
+    if (walletCapabilities.hasAtomicBatch) {
```

Comment:
should this use batchedTransactions even only one transaction?

Reply:
yep - batch of 1 is valid as well